csharp/src/Microsoft.ML.OnnxRuntime/Tensors/Tensor.cs

```diff
@@ -397,15 +397,19 @@ protected Tensor(int length) : base(typeof(T))
         }
 
         /// <summary>
-        /// Initialize an n-dimensional tensor with the specified dimensions and layout.  ReverseStride=true gives a stride of 1-element witdth to the first dimension (0).  ReverseStride=false gives a stride of 1-element width to the last dimension (n-1).
+        /// Initialize an n-dimensional tensor with the specified dimensions and layout.  ReverseStride=true gives a stride of 1-element width to the first dimension (0).  ReverseStride=false gives a stride of 1-element width to the last dimension (n-1).
         /// </summary>
         /// <param name="dimensions">An span of integers that represent the size of each dimension of the Tensor to create.</param>
         /// <param name="reverseStride">False (default) to indicate that the first dimension is most major (farthest apart) and the last dimension is most minor (closest together): akin to row-major in a rank-2 tensor.  True to indicate that the last dimension is most major (farthest apart) and the first dimension is most minor (closest together): akin to column-major in a rank-2 tensor.</param>
         protected Tensor(ReadOnlySpan<int> dimensions, bool reverseStride) : base(typeof(T))
         {
+            // Scalar tensor
             if (dimensions.Length == 0)
```

Comment:
> if (dimensions.Length == 0) [](start = 11, length = 28)

Should we add else clause to it? Everything done here is overwritten below

Reply:
Oh yeah. looks like I forgot to copy over the else when I copied this piece of code from the original PR Actually - we may consider dropping this code altogether - looks like what exists can handle scalars - no need to special case it.

Reply:
Took out the entire "special case" logic. The existing code can handle it as such. Just removed the throwing exception part.